lib/bootsnap/load_path_cache/core_ext/kernel_require.rb

```diff
@@ -13,72 +13,106 @@ def self.make_load_error(path)
 module Kernel
   alias_method :require_without_cache, :require
   def require(path)
+    return false if Bootsnap::LoadPathCache.loaded_features_index.key?(path)
     if resolved = Bootsnap::LoadPathCache.load_path_cache.find(path)
-      require_without_cache(resolved)
+      Bootsnap::LoadPathCache.load_path_cache.register(path, resolved) do
```

Comment:
Should no it be `loaded_features_index` instead of `load_path_cache`?

Reply:
whoah, yeah, nice catch -- thanks